script/make_release.py

```diff
@@ -0,0 +1,330 @@
+        self.path = prefix
+        if add_uuid:
+            self.path += str(uuid.uuid4())
+        os.mkdir(self.path)
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        shutil.rmtree(self.path, onerror=remove_readonly)
+
+def get_bench_from_git_history(repo_dir):
+    command = ['git', '--git-dir', os.path.join(repo_dir, '.git'), 'log']
+    completed_process = subprocess.run(command, text=True, capture_output=True, universal_newlines=True)
+    output = completed_process.stdout
+    try:
+        m = re.search('[Bb]ench[ :]+(\\d*)', output)
+        return int(m.group(1))
+    except:
+        return None
+
+def pack_binary_with_source(dest_dir, repo_dir, src_file, dest_filename):
+    '''
+    Takes the src_file (expected to be stockfish's binary) and
+    the `repo_dir` and produces a .zip archive with them both.
+    The archive is put into `dest_dir` and has the following structure:
+    `artifact_name` is the stem of `dest_filename`
+    - `dest_dir`
+        - `artifact_name`+ZIP_SUFFIX
+            - `artifact_name`
+                - `Stockfish`
+                    - `src`
+                    - rest of the repo, without .git
+                - `artifact_name`+EXE_SUFFIX
+    '''
+
+    artifact_name = os.path.splitext(dest_filename)[0]
+    archive_filename = artifact_name + ZIP_SUFFIX
+    archive_path = os.path.join(dest_dir, archive_filename)
+
+    LOGGER.info('Creating archive {}.'.format(archive_path))
+
+    with ZipFile(archive_path, 'w') as archive:
+        # Write the stockfish binary, AUTHORS, and Copying.txt files
+        # to the main directory.
+        main_level_files = [
+            (src_file, dest_filename),
+            (os.path.join(repo_dir, 'AUTHORS'), 'AUTHORS'),
+            (os.path.join(repo_dir, 'Copying.txt'), 'Copying.txt')
+        ]
+        for path_from, name in main_level_files:
+            archive.write(path_from, os.path.join(artifact_name, name))
+
+        # Write the source directory, excluding .git.
+        for path, subdirs, files in os.walk(repo_dir, followlinks=False):
+            if '.git' in path:
+                continue
+
+            for filename in files:
+                full_path = os.path.join(path, filename)
+                rel_path = os.path.relpath(full_path, repo_dir)
+                internal_path = os.path.join(artifact_name, 'Stockfish', rel_path)
+
+                archive.write(full_path, internal_path)
+
+    success = True
+    with ZipFile(archive_path, 'r') as archive:
+        if archive.testzip() is not None:
+            success = False
+
+    if not success:
+        os.remove(archive_path)
+        raise Exception('The CRC check for {} failed.'.format(archive_path))
+
+    LOGGER.info('Created archive {}.'.format(archive_path))
+
+def run_bench(stockfish):
+    command = [stockfish, 'bench']
+    if SDE_PATH:
+        command = [SDE_PATH, '--'] + command
+
```

Comment:
I reworked a bit also the list comprehension.
Anyway I used the comprehension because I'm experimenting with the cli, feel free to rework the code :)